src/Pact/Native.hs

```diff
@@ -1338,9 +1342,13 @@ strToInt i as =
     doBase si base txt = case baseStrToInt base txt of
       Left e -> evalError' si (pretty e)
       Right n -> return (toTerm n)
-    doBase64 si txt = case parseB64UrlUnpaddedText' txt of
-      Left e -> evalError' si (pretty e)
-      Right bs -> return $ toTerm $ bsToInteger bs
+    doBase64 si txt = do
+      -- Use Legacy error behavior when 4.9 is disabled.
+      base64Behavior <- bool Simplified Legacy <$> isExecutionFlagSet FlagDisablePact49
```

Comment:
Let's move this boolean check into `base64DecodeWithShimmedErrors`

Reply:
:+1: Much better.